src/nodes/primitive/range.ts

```diff
@@ -69,6 +74,9 @@ export const invertedComparators = {
 
 export type InvertedComparators = typeof invertedComparators
 
+export const writeIncompatibleLimitMessage = (l: string, r: string) =>
```

Comment:
We could probably accept a more specific type like `number` | `Date`. If you wanted to get fancy, you could even have `l` be generic, then have `r` be the opposite type of `l`